src/lib.rs

```diff
@@ -445,6 +450,12 @@ impl fmt::Display for Error {
             Error::IncorrectPubkeyHash => {
                 f.write_str("Incorrect pubkey hash for given descriptor pkh/wpkh")
             }
+            Error::UnexpectedPubkeyHash => {
+                f.write_str("Was expecting public key, not a public key hash")
+            }
+            Error::PubkeyProcessorFailure => {
+                f.write_str("Internal fialure of the public key processor")
```

Comment:
I don't think any of these error types are needed anymore.

Reply:
Correct, will remove them